lib/3scale/backend/stats/codes_commons.rb

```diff
@@ -4,13 +4,13 @@ module Stats
       module CodesCommons
         TRACKED_CODES = [200, 404, 403, 500, 503].freeze
 
-        HTTP_CODE_GROUPS_MAP = TRACKED_CODES.each_with_object({}) do |code, hsh|
-          hsh[code / 100] = "#{code / 100}XX"
-        end
-
         def self.get_http_code_group(http_code)
-          HTTP_CODE_GROUPS_MAP.fetch(http_code / 100)
+          "#{http_code / 100}XX"
         end
+
+        TRACKED_CODE_GROUPS = TRACKED_CODES.each_with_object({}) do |code, group|
```

Comment:
I'd change this to `TRACKED_CODE_GROUPS = ['2xx', '4xx', '5xx']` for simplicity given that the set of `TRACKED_CODES` is unlikely to change.